### PR TITLE
Add CPU info in the check update user-agent

### PR DESCRIPTION
**Pull request by vadmeste**
## Description
Example, Add this to the existing user-agent header:

```
   CPU (total_cpus:1, total_cores:6; vendor:GenuineIntel; family:6; model:167; stepping:1; model_name:Intel(R) Xeon(R) E-2386G CPU @ 3.50GHz)```

```
## Types of changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Optimization (provides speedup with no functional changes)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:
- [ ] Fixes a regression (If yes, please add `commit-id` or `PR #` here)
- [ ] Unit tests added/updated
- [ ] Internal documentation updated
- [ ] Create a documentation update request [here](https://github.com/minio/docs/issues/new?label=doc-change,title=Doc+Updated+Needed+For+PR+github.com%2fminio%2fminio%2fpull%2fNNNNN)

### CODE REVIEW

**Comment by minio-trusted** [comment]
## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-large-bucket.sh | :heavy_check_mark: |
| mint-fs.sh | :heavy_check_mark: |
| mint-erasure.sh | :heavy_check_mark: |
| mint-dist-erasure.sh | :heavy_check_mark: |
| mint-compress-encrypt-dist-erasure.sh | :heavy_check_mark: |
| mint-pools.sh | :heavy_check_mark: |
Deleting image on docker hub
Deleting image locally

**Comment by minio-trusted** [comment]
## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-large-bucket.sh | :heavy_check_mark: |
| mint-fs.sh | :heavy_check_mark: |
| mint-erasure.sh | :heavy_check_mark: |
| mint-dist-erasure.sh | :heavy_check_mark: |
| mint-compress-encrypt-dist-erasure.sh | :heavy_check_mark: |
| mint-pools.sh | :heavy_check_mark: |
Deleting image on docker hub
Deleting image locally

**Comment by minio-trusted** [comment]
## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-large-bucket.sh | :heavy_check_mark: |
| mint-fs.sh | :heavy_check_mark: |
| mint-erasure.sh | :heavy_check_mark: |
| mint-dist-erasure.sh | :heavy_check_mark: |
| mint-compress-encrypt-dist-erasure.sh | :heavy_check_mark: |
| mint-pools.sh | :heavy_check_mark: |
Deleting image on docker hub
Deleting image locally

**Comment by minio-trusted** [comment]
## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-large-bucket.sh | :heavy_check_mark: |
| mint-fs.sh | :heavy_check_mark: |
| mint-erasure.sh | :heavy_check_mark: |
| mint-dist-erasure.sh | :heavy_check_mark: |
| mint-compress-encrypt-dist-erasure.sh | :heavy_check_mark: |
| mint-pools.sh | :heavy_check_mark: |
Deleting image on docker hub
Deleting image locally

File: cmd/update.go [hunk 1]
```diff
@@ -277,6 +278,9 @@ func getUserAgent(mode string) string {
 		}
 	}
 
+	uaAppend(" CPU ", fmt.Sprintf("(total_cores:%d; vendor:%s; family:%d; model:%d; stepping:%d; model_name:%s)",
+		runtime.NumCPU()*cpuid.CPU.PhysicalCores, cpuid.CPU.VendorString, cpuid.CPU.Family, cpuid.CPU.Model, cpuid.CPU.Stepping, cpuid.CPU.BrandName))
```
Comment (klauspost) [inline cmd/update.go]: `runtime.NumCPU()*cpuid.CPU.PhysicalCores` for sure isn't correct. `runtime.NumCPU()` returns the number of "logical CPUs", ie execution threads - presumably across all physical cpus in total.

Example:
```
runtime.NumCPU(): 32

cpuid.CPU.BrandName: AMD Ryzen 9 3950X 16-Core Processor
cpuid.CPU.PhysicalCores: 16
cpuid.CPU.ThreadsPerCore: 2
cpuid.CPU.LogicalCores: 32
```

I think you just want `runtime.NumCPU()`.

**Review by klauspost — APPROVED** [review]

**Comment by minio-trusted** [comment]
## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-large-bucket.sh | :heavy_check_mark: |
| mint-fs.sh | :heavy_check_mark: |
| mint-erasure.sh | :heavy_check_mark: |
| mint-dist-erasure.sh | :heavy_check_mark: |
| mint-compress-encrypt-dist-erasure.sh | :heavy_check_mark: |
| mint-pools.sh | :heavy_check_mark: |
Deleting image on docker hub
Deleting image locally

**Comment by vadmeste** [comment]
Good catch @klauspost. I made one more change to detect the number of CPUs and the number of physical cores.

**Comment by minio-trusted** [comment]
## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-large-bucket.sh | :heavy_check_mark: |
| mint-fs.sh | :heavy_check_mark: |
| mint-erasure.sh | :heavy_check_mark: |
| mint-dist-erasure.sh | :heavy_check_mark: |
| mint-compress-encrypt-dist-erasure.sh | :heavy_check_mark: |
| mint-pools.sh | :heavy_check_mark: |
Deleting image on docker hub
Deleting image locally

**Comment by harshavardhana** [comment]
@vadmeste you may want to fix the tests

```
--- FAIL: TestUserAgent (0.00s)
    update_test.go:183: Test 1: expected: MinIO (linux; amd64; mode-server-fs; source) MinIO/DEVELOPMENT.GOGET MinIO/DEVELOPMENT.GOGET MinIO/DEVELOPMENT.GOGET, got: MinIO (linux; amd64; mode-server-fs; source) MinIO/DEVELOPMENT.GOGET MinIO/DEVELOPMENT.GOGET MinIO/DEVELOPMENT.GOGET CPU (total_cpus:1, total_cores:2; vendor:GenuineIntel; family:6; model:106; stepping:6; model_name:Intel(R) Xeon(R) Platinum 8370C CPU @ 2.80GHz)
    update_test.go:183: Test 2: expected: MinIO (linux; amd64; mode-server-xl; dcos; source) MinIO/DEVELOPMENT.GOGET MinIO/DEVELOPMENT.GOGET MinIO/DEVELOPMENT.GOGET MinIO/universe-mesos-1111, got: MinIO (linux; amd64; mode-server-xl; dcos; source) MinIO/DEVELOPMENT.GOGET MinIO/DEVELOPMENT.GOGET MinIO/DEVELOPMENT.GOGET MinIO/universe-mesos-1111 CPU (total_cpus:1, total_cores:2; vendor:GenuineIntel; family:6; model:106; stepping:6; model_name:Intel(R) Xeon(R) Platinum 8370C CPU @ 2.80GHz)
    update_test.go:183: Test 3: expected: MinIO (linux; amd64; mode-server-xl; kubernetes; source) MinIO/DEVELOPMENT.GOGET MinIO/DEVELOPMENT.GOGET MinIO/DEVELOPMENT.GOGET, got: MinIO (linux; amd64; mode-server-xl; kubernetes; source) MinIO/DEVELOPMENT.GOGET MinIO/DEVELOPMENT.GOGET MinIO/DEVELOPMENT.GOGET CPU (total_cpus:1, total_cores:2; vendor:GenuineIntel; family:6; model:106; stepping:6; model_name:Intel(R) Xeon(R) Platinum 8370C CPU @ 2.80GHz)
FAIL
```

**Review by harshavardhana — APPROVED** [review]

**Review by harshavardhana — APPROVED** [review]

**Comment by minio-trusted** [comment]
## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-large-bucket.sh | :heavy_check_mark: |
| mint-fs.sh | :heavy_check_mark: |
| mint-erasure.sh | :heavy_check_mark: |
| mint-dist-erasure.sh | :heavy_check_mark: |
| mint-compress-encrypt-dist-erasure.sh | :heavy_check_mark: |
| mint-pools.sh | :heavy_check_mark: |
Deleting image on docker hub
Deleting image locally

**Pull request merged**